core/pkg/eval/json_evaluator_test.go

```diff
@@ -1207,3 +1207,45 @@ func TestFlagStateSafeForConcurrentReadWrites(t *testing.T) {
 		})
 	}
 }
+
+func TestFlagdAmbientProperties(t *testing.T) {
+	t.Run("flagKeyIsInTheContext", func(t *testing.T) {
+		evaluator := eval.NewJSONEvaluator(logger.NewLogger(nil, false), store.NewFlags())
+
+		_, _, err := evaluator.SetState(sync.DataSync{FlagData: `{
+			"flags": {
+				"welcome-banner": {
+					"state": "ENABLED",
+					"variants": {
+						"true": true,
+						"false": false
+					},
+					"defaultVariant": "false",
+					"targeting": {
+						"==": [ { "var": "$flagd.flag_key" }, "welcome-banner" ]
```

Comment:
I think this needs to be changed to match the field name of the `flagdProperties` struct
```suggestion
						"==": [ { "var": "$flagd.FlagKey" }, "welcome-banner" ]
```

Reply:
Yes, I think this is true... however, I'm not sure we want to use this casing. All implementations will have to populate this prop, and in that case, I would advocate for something more in line with start JSON casing (ie: camelCase), so `flagKey`.

What do you folks think? @craigpastro @bacherfl @beeme1mr

Reply:
I agree, `flagKey` would be more in line with typical json casing

Reply:
Ah, silly me. I updated last night and failed to update the test. I also like camelCase for JSON fields so I'll update to `flagKey`.